Kernel/Tasks/Process.cpp

```diff
@@ -173,28 +173,27 @@ UNMAP_AFTER_INIT void Process::initialize()
 
 void Process::kill_threads_except_self()
 {
-    InterruptDisabler disabler;
-
-    if (thread_count() <= 1)
-        return;
-
-    auto* current_thread = Thread::current();
-    for_each_thread([&](Thread& thread) {
-        if (&thread == current_thread)
-            return;
-
-        if (auto state = thread.state(); state == Thread::State::Dead
-            || state == Thread::State::Dying)
+    m_thread_count.with([this](auto& count) {
```

Comment:
here, a comment to say that we lock the thread count so that no one can create new threads while we are finalizing would be good. Are there any other safeguards against that?